pandas/core/arrays/string_arrow.py

```diff
@@ -758,6 +759,69 @@ def _str_map(self, f, na_value=None, dtype: Dtype | None = None):
             # -> We don't know the result type. E.g. `.get` can return anything.
             return lib.map_infer_mask(arr, f, mask.view("uint8"))
 
+    def _str_isalnum(self):
+        if hasattr(pc, "utf8_is_alnum"):
+            result = pc.utf8_is_alnum(self._data)
```

Comment:
At some point (not necessarily this PR), it might be worth benchmarking to see if calling `pc.string_is_ascii` first to then potentially use `pc.ascii_is_alnum` instead of `pc.utf8_is_alnum` could be worth it (which would be assuming that testing whether it's all ascii takes much less time than the benefit from using the faster ascii algorithm vs the utf8 one)